source/deploy/deployed-systems.rst

```diff
@@ -0,0 +1,199 @@
+
+Load balancing
+--------------
+
+For sites that need load balancing, we typically
+use AWS Elastic Load Balancer. We put the public SSL certificate
+on the load balancers, have ELB accept both port 80 and port 443
+requests, and forward them on the corresponding port to our
+backend servers.
+
+For sites that force SSL, the backend servers are responsible
+for responding to port 80 requests with a redirect to the SSL
+URL of the site.
+
+When not on AWS, we've used haproxy for load balancing and
+failover.
+
+Web server
+----------
+
+Our servers use `nginx` to receive incoming requests. Requests
+not addressed to a known domain are failed immediately with a
+444 status. Static files are served directly by nginx. Other
+requests are proxied to gunicorn.
+
+For sites that force SSL, requests to port 80 are redirected to
+the corresponding ``https`` URL by nginx, and nginx adds headers such as
+``Strict-Transport-Security``.
+
+Static files
+------------
+
+For most projects, we serve our static files using Nginx. On
+some, we store static and media files on S3 and let AWS serve
+them.
+
+`This blog post <https://www.caktusgroup.com/blog/2014/11/10/Using-Amazon-S3-to-store-your-Django-sites-static-and-media-files/>`_
+has more details.
+
+Django HTTP service
+-------------------
+
+Django processes that serve HTTP requests are run using
+gunicorn.  Nginx proxies requests to gunicorn over a local
+TCP or Unix socket.
+
+Asynchronous work
+~~~~~~~~~~~~~~~~~
+
+Django workers
+--------------
+
+For asynchronous work, we use Celery's worker daemon.
+
+Queue
+-----
+
+We use rabbitmq to queue asynchronous work.
+
+Misc. services
+~~~~~~~~~~~~~~
+
+Caching
+-------
+
+Depending on the project, we might use memcached, redis,
+or both for caching.
+
+memcached is good for caching responses and data.
+
+redis is good for a session store, since it's fast but
+persistent.
+
+Database server
+---------------
+
+We always use PostgreSQL as our database. On most sites, we run the
+server on one system. On sites with high load or that need high
+availability, we run replicas on other systems, and can send read-only
+database requests to them to relieve traffic on the master server.
+
+In some projects, we use Amazon RDS to host the PostgreSQL service.
+
+Logging
+-------
+
+We are moving toward consolidating all of our logging into the
+local syslog, and then forwarding it to a log monitoring
+service such as Papertrail. That both lets us set up alerts
+that notify us when certain things show up in the logs,
+and to have a central place to go where we can view and
+search all the logs.
+
+For processes managed by supervisor, we just configure them
+to log to stdout, then have supervisor pass that to syslog.
+For other processes, we either configure them to log to
+syslog, or use rsyslog's `imfile` input plugin to monitor
+their text log files and pull them into syslog.
+
+Then we can configure rsyslog to forward log messages
+to Papertrail.
```

Comment:
This whole document is about how the system ends up configured, not how to run the current deploy system, so I think this is okay as-is.